### ISSUE TITLE
[Responsive Navigation] A Couple Issues with Arrows...

### ISSUE THREAD

**Issue by Lynda333** [zurb/foundation-sites#7095]
I've noticed a couple issues that I have with the way arrows are placed with the responsive navigation. 

First let's take a look at a menu I have setup: 

![dropdown](https://cloud.githubusercontent.com/assets/1938824/11316082/7729fe1a-8fbb-11e5-82d1-70898b6ed4d3.png)

(For reference the code for my menu is at the bottom of this post.)

I have a dropdown menu setup that turns into a drilldown menu when on a small screen. Works great but the arrows are my issue. 

First the arrow points to the right. When the menu is horizontal the arrow should point down. I see this as a UX problem myself. If the menu was vertical or even in drilldown mode, then this would be acceptable.

Second, I am using the `expanded` class for a nicely spaced menu. I love the class but I prefer the arrows to be next to the words, not all the way to the right. Depending upon the size of the menu, the arrows are placed to far from the words they go with.

Here is a screenshot for this problem: 

![arrows](https://cloud.githubusercontent.com/assets/1938824/12870752/754f3370-cd10-11e5-8fb7-2cd4266338af.jpg)

I assume some changes to the SCSS is all that is needed. I believe I can manually override the first issue with some media queries (haven't tried yet) but I am unsure how to fix the second issue.

``` html
<div class="title-bar" data-responsive-toggle="mainMenu" data-hide-for="medium">
  <button class="menu-icon" type="button" data-toggle></button>
  <div class="title-bar-title">Menu</div>
</div>

<ul class="vertical medium-horizontal menu expanded" data-responsive-menu="drilldown medium-dropdown" id="mainMenu">
  <li><a href="#">Home</a></li>
  <li class="has-submenu"><a href="#">News</a>
    <ul class="vertical submenu menu" data-submenu>
      <li><a href="#">Colorado News</a></li>
      <li><a href="#">National &amp; International News</a></li>
    </ul>
  </li>
  <li class="has-submenu"><a href="#">Information</a>
    <ul class="vertical submenu menu" data-submenu>
      <li><a href="#">City &amp; County Regulations</a></li>
      <li><a href="#">Denver Information</a></li>
      <li><a href="#">General Information</a></li>
      <li><a href="#">Case Law</a></li>
      <li><a href="#">Federal Letters</a></li>
      <li><a href="#">Colorado Patient Services</a></li>
      <li><a href="#">Power Point Presentations</a></li>
    </ul>
  </li>
  <li><a href="#">Blog</a></li>
  <li><a href="#">Media</a></li>
  <li><a href="#">About</a></li>
  <li><a href="#">Contact</a></li>
</ul>
```

**Comment by aaw922** [comment]
Depending on the menu, you may adjust the arrows distance by increasing or decreasing the property 'right: px' for said class.  

In your particular scenario I believe the class to edit is: 

.dropdown.menu .is-dropdown-submenu-parent.is-down-arrow > a::after {
    content: '';
    display: block;
    width: 0;
    height: 0;
    border: inset 5px;
    border-color: #2199e8 transparent transparent;
    border-top-style: solid;
    position: absolute;
    top: 0.825rem;
    right: 5px;   <--- Property to Edit.
}
You will most likely see a larger number in your example. Increasing the value of this property should bring the arrow closer to the text before it.

Test results -
Arrows pushed away from text
![arrow-distance-far](https://cloud.githubusercontent.com/assets/9277568/12012469/c7e1290c-acba-11e5-81e2-73db0336f3d0.png)
.  
Property value for arrows being pushed away.
![property-value-far](https://cloud.githubusercontent.com/assets/9277568/12012471/decf2696-acba-11e5-84fc-8163beee05f8.png)

Arrows close to text.
![arrow-distance-close](https://cloud.githubusercontent.com/assets/9277568/12012472/e92c2d78-acba-11e5-965d-7221142694e7.png)

Property value of arrows close to text.
![property-value-close](https://cloud.githubusercontent.com/assets/9277568/12012473/f2f5870a-acba-11e5-85c9-48f45f945a8e.png)

**Comment by rafibomb** [comment]
On your `<ul>` remove the `.vertical` and `.medium-horizontal` to fix the arrows. The data attribute takes care of the arrows in the JS.

**Comment by Lynda333** [comment]
@rafibomb - You're correct about that. Maybe provide some examples that are a bit more clear? 

The arrows with the `expanded` class are still an issue.

**Comment by purplespider** [comment]
I've just stumbled across this issue too. See this example: http://codepen.io/purplespider/pen/pgqOdZ

I'm just after an accordion menu on small screens, and a horizontal drop-down menu on medium and up.

In it's current state the dropdown arrows point to the right, instead of down:
![](http://spdr.me/1clfe+)
Small:
![](http://spdr.me/16bdG+)


If you remove `.vertical` and `.medium-horizontal` from the `<ul>` it fixes the arrow issue on medium screens: 
![](http://spdr.me/13Hde+)
But the accordion menu breaks on small screens as it's now not being told to display vertically:
![](http://spdr.me/O3AK+)

Seems like a bug to me.

**Comment by Lynda333** [comment]
@rafibomb - As @purplespider pointed out, when you remove the vertical classes, it's broken on small screens. It's more than just arrows pointing the wrong way.

At this point I'm going to use show/hide classes to show my horizontal menu on desktop and then hide that menu and have a duplicate menu for small. Only way I see to do this until this bug is fixed.

**Comment by gakimball** [comment]
The issues with arrow direction have been resolved in Foundation 6.2.

As for the `.expanded` issue, we're going to chalk that up to a design choice. The arrows are `position: absolute`, so if you want to move them around, you can change them to `position: static` and `display: inline-block`. Then they should sit right next to the words.
